src/Microsoft.Data/MutableDataFrameBuffer.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A basic mutable store to hold values in a DataFrame column. Supports wrapping with an ArrowBuffer
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    internal class MutableDataFrameBuffer<T> : DataFrameBuffer<T>
```

Comment:
I would go the opposite route with naming. We have `Span` and `ReadOnlySpan`, and `Memory` and `ReadOnlyMemory`. We should follow the same naming pattern.